clap_builder/src/derive.rs

```diff
@@ -387,3 +388,64 @@ fn format_error<I: CommandFactory>(err: Error) -> Error {
     let mut cmd = I::command();
     err.format(&mut cmd)
 }
+
+impl FromArgMatches for () {
+    fn from_arg_matches(_matches: &ArgMatches) -> Result<Self, Error> {
+        Ok(())
+    }
+
+    fn update_from_arg_matches(&mut self, _matches: &ArgMatches) -> Result<(), Error> {
+        Ok(())
+    }
+}
+
+impl Args for () {
+    fn augment_args(cmd: Command) -> Command {
+        cmd
+    }
+
+    fn augment_args_for_update(cmd: Command) -> Command {
+        cmd
+    }
+}
+
+impl Subcommand for () {
+    fn augment_subcommands(cmd: Command) -> Command {
+        cmd
+    }
+
+    fn augment_subcommands_for_update(cmd: Command) -> Command {
+        cmd
+    }
+
+    fn has_subcommand(_name: &str) -> bool {
+        false
+    }
+}
+
+impl FromArgMatches for Infallible {
+    fn from_arg_matches(_matches: &ArgMatches) -> Result<Self, Error> {
+        Err(Error::raw(
+            crate::error::ErrorKind::MissingSubcommand,
+            "A subcommand is required but one was not provided.",
```

Comment:
Our error messages should be lower case.  I found a couple that aren't and am preparing a separate PR for those.

Reply:
Since this is directly copied from (the derive output)[https://github.com/clap-rs/clap/blob/d045daa5c0c096280d4ff983915e56105d42cf4f/clap_derive/src/derives/subcommand.rs#L568] should I change it or leave as is for now?

If it's lower case then also without the dot at the end?